src/transformers/modeling_flax_bert.py

```diff
@@ -0,0 +1,302 @@
+
+    def apply(self, hidden_state, attention_mask, num_heads: int, head_size: int, intermediate_size: int):
+        attention = BertAttention(hidden_state, attention_mask, num_heads, head_size, intermediate_size, name="attention")
+        intermediate = BertIntermediate(attention, intermediate_size, name="intermediate")
+        output = BertOutput(intermediate, attention, name="output")
+
+        return output
+
+
+class BertLayerCollection(nn.Module):
+    """
+    Stores N BertLayer(s)
+    """
+    def apply(self, input, attention_mask, num_layers: int, num_heads: int, head_size: int, intermediate_size: int):
+        assert num_layers > 0, "num_layers should be >= 1, got ({})".format(num_layers)
+
+        # Initialize input / output
+        input_i = output_i = input
+
+        # Forward over all encoders
+        for i in range(num_layers):
+            output_i = BertLayer(input_i, attention_mask, num_heads, head_size, intermediate_size, name="{}".format(i))
+            input_i = output_i
+        return output_i
+
+
+class BertEncoder(nn.Module):
+
+    def apply(self, hidden_state, attention_mask, num_layers: int, num_heads: int, head_size: int, intermediate_size: int):
+        encodings = BertLayerCollection(hidden_state, attention_mask, num_layers, num_heads, head_size, intermediate_size, name="layer")
+        return encodings
+
+
+class BertPooler(nn.Module):
+
+    def apply(self, hidden_state):
+        first_token = hidden_state[:, 0]
+        out = nn.Dense(first_token, hidden_state.shape[-1], name="dense")
+        return jax.lax.tanh(out)
+
+
+class BertModel(nn.Module):
+
+    def apply(self, input_ids, token_type_ids, position_ids, attention_mask,
+              vocab_size: int, hidden_size: int, type_vocab_size: int,
+              max_length: int, num_encoder_layers: int, num_heads: int,
+              head_size: int, intermediate_size: int, padding_idx: int,
+              emb_init: Callable[..., np.ndarray] = nn.initializers.normal(stddev=0.1)):
+
+        # Embedding
+        embeddings = BertEmbeddings(
+            input_ids, token_type_ids, position_ids, attention_mask,
+            vocab_size, hidden_size, type_vocab_size, max_length,
+            name="embeddings"
+        )
+
+        # N stacked encoding layers
+        encoder = BertEncoder(
+            embeddings, attention_mask, num_encoder_layers,
+            num_heads, head_size, intermediate_size,
+            name="encoder"
+        )
+
+        pooled = BertPooler(encoder, name="pooler")
+        return encoder, pooled
+
+
+class FlaxBertModel(JaxPreTrainedModel):
+    """
+    BERT implementation using JAX/Flax as backend
+    """
+
+    model_class = BertModel
+    config_class = BertConfig
+    pretrained_model_archive_map = BERT_PRETRAINED_MODEL_ARCHIVE_MAP
+    base_model_prefix = "bert"
+
+    def __init__(self, config: BertConfig, state: dict, **kwargs):
+        self.config = config
+        self.key = PRNGKey(0)
```

Comment:
What is this used for?

Reply:
It's mainly usage is related to all the stochastic operations, such as Dropout that I didn't put into the model for now, but might be pushed soon.